tdm-be/src/lib/cron.ts

```diff
@@ -0,0 +1,66 @@
+import environment from '~/lib/config';
+import { customFile, filesLocation } from '~/lib/files';
+import { cronLogger } from '~/lib/logger';
+
+import cron from 'node-cron';
+
+import fs from 'node:fs/promises';
+
+const createFileCron = async (directory: string) => {
+    const oneWeekAgo = new Date(); // Date actuelle
+    oneWeekAgo.setDate(oneWeekAgo.getDate() - environment.cron.deleteFileOlderThan);
+
+    const files = await fs.readdir(customFile(directory));
+
+    for (const file of files) {
+        const filePath = customFile(directory, file);
+        const stats = await fs.stat(filePath);
+
+        if (stats.isFile() && stats.mtime < oneWeekAgo) {
+            await fs.unlink(filePath);
+            cronLogger.info(`${filePath} has been delete`);
+        }
+    }
+};
+
+const initCron = () => {
+    cronLogger.debug('Creating upload file cron');
+    cron.schedule(environment.cron.schedule, () => {
+        createFileCron(filesLocation.upload)
+            .then(() => {
+                cronLogger.info('Upload file cron ended successfully');
+            })
+            .catch((reason) => {
+                cronLogger.error('Upload file cron ended with a crash');
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏Rather say file*s* than file: it's likely to be several files
```suggestion
    cronLogger.debug('Creating upload files cron');
    cron.schedule(environment.cron.schedule, () => {
        createFileCron(filesLocation.upload)
            .then(() => {
                cronLogger.info('Upload files cron ended successfully');
            })
            .catch((reason) => {
                cronLogger.error('Upload files cron ended with a crash');
```